src/grammar_optimiser/extend_grammar.jl

```diff
@@ -0,0 +1,93 @@
+"""
+    generate_tree_from_compression(parent, d, compression_id, grammar)
+
+Generates a Herb tree from a given compression.
```

Comment:
Herb tree -> RuleNode. You can also set references in docstrings. Example from HerbCore:

`Create a [`RuleNode`](@ref) for the [`AbstractGrammar`](@ref) `